mirobo/ceil.py

```diff
@@ -0,0 +1,112 @@
+from .device import Device
+from typing import Any, Dict
+
+
+class Ceil(Device):
+    """Main class representing Xiaomi Philips LED Ceiling Lamp."""
+
+    # TODO: - Auto On/Off Not Supported
+    #       - Adjust Scens with Wall Switch Not Supported
+
+    def on(self):
+        """Power on."""
+        return self.send("set_power", ["on"])
+
+    def off(self):
+        """Power off."""
+        return self.send("set_power", ["off"])
+
+    def set_bright(self, level: int):
+        """Set brightness level."""
+        return self.send("set_bright", [level])
+
+    def set_cct(self, level: int):
+        """Set Correlated Color Temperature."""
+        return self.send("set_cct", [level])
+
+    def delay_off(self, seconds: int):
+        """Set delay off seconds."""
+        return self.send("delay_off", [seconds])
+
+    def set_scene(self, num: int):
+        """Set scene number."""
+        return self.send("apply_fixed_scene", [num])
+
+    def bl_on(self):
+        """Smart Midnight Light On."""
+        return self.send("enable_bl", [1])
+
+    def bl_off(self):
+        """Smart Midnight Light off."""
+        return self.send("enable_bl", [0])
+
+    def ac_on(self):
+        """Auto CCT On."""
+        return self.send("enable_ac", [1])
+
+    def ac_off(self):
+        """Auto CCT Off."""
+        return self.send("enable_ac", [0])
+
+    def status(self):
```

Comment:
This could be set with typing to return `CeilStatus`, or is that deliberately left out to avoid moving it on top of this class?